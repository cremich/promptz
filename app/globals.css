pre,
code {
  word-wrap: break-word !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  margin: 0;
  padding: 0;
}

code {
  background-color: #f3f3f7;
}

pre > code {
  display: inline-block;
  padding: 1em;
}
