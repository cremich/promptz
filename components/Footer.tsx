// components/TopNav.tsx

"use client";

import {
  Box,
  ColumnLayout,
  Link,
  SpaceBetween,
} from "@cloudscape-design/components";

export default function Footer() {
  return (
    <div>
      <div style={{ backgroundColor: "#f9f9fa" }}>
        <Box margin={"xxl"} textAlign="center">
          <ColumnLayout columns={3}>
            <Box textAlign="center">
              <h3>Why Use PROMPTZ?</h3>
              Simplify software development with ready-to-use solutions.
              Collaborate with the community to refine and share your ideas.
              Learn prompt engineering best practices for every stage of the
              SDLC.
            </Box>
            <Box textAlign="center">
              <h3>About PROMPTZ</h3>
              <p>
                PROMPTZ is your ultimate resource for Amazon Q Developer prompt
                engineering. Discover, create, and share high-quality prompts to
                tackle real-world software development challenges. From
                generating AWS architecture blueprints to automating workflows
                and beyond, PROMPTZ empowers developers to get the most out of
                Amazon Q Developer with a rich library of prompt templates and
                community contributions.
              </p>
            </Box>
            <Box textAlign="center">
              <h3>Contribute to PROMPTZ</h3>
              PROMPTZ is a community-first platform. The source code is open on
              GitHub, and we welcome contributions from developers and
              enthusiasts. If you want to share your ideas, join us on{" "}
              <a href="https://github.com/cremich/promptz">Github</a>.
            </Box>
          </ColumnLayout>
        </Box>
      </div>
      <Box textAlign="center" margin={"xxl"}>
        <SpaceBetween size="m">
          <div>
            Stay connected with PROMPTZ! ‚≠ê us on{" "}
            <Link
              external
              href="https://github.com/cremich/promptz"
              variant="secondary"
            >
              Github
            </Link>{" "}
            and join the conversation.
          </div>
          <div>
            Made with üíú by{" "}
            <Link
              external
              href="https://www.linkedin.com/in/christian-bonzelet/"
              variant="secondary"
            >
              Christian Bonzelet
            </Link>{" "}
            with a lot of help by{" "}
            <Link
              external
              href="https://aws.amazon.com/q/developer/"
              variant="secondary"
            >
              Amazon Q Developer
            </Link>
          </div>
        </SpaceBetween>
      </Box>
    </div>
  );
}
