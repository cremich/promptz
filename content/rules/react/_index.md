---
title: "React"
description: "Rules for React development, component patterns, and best practices."
---
