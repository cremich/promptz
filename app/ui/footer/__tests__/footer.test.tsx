import { describe, expect, test } from "@jest/globals";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Footer from "@/app/ui/footer/footer";

describe("Footer", () => {
  test("renders footer unchanged", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
