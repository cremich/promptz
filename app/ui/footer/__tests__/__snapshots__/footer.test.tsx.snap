// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footer unchanged 1`] = `
<div>
  <footer
    class="w-full border-t mt-12"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex flex-col items-center justify-between py-8 sm:flex-row"
      >
        <p
          class="text-sm text-gray-400"
        >
          Â© PROMPTZ. All rights reserved. - Made by
           
          <a
            class="text-gray-400 hover:text-violet-500 underline"
            href="https://linkedin.com/in/christian-bonzelet"
            rel="noopener noreferrer"
            target="_blank"
          >
            Christian Bonzelet
          </a>
           
          with a lot of help by
           
          <a
            class="text-gray-400 hover:text-violet-500 underline"
            href="https://aws.amazon.com/q/developer/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Amazon Q Developer
          </a>
        </p>
        <div
          class="flex items-center gap-4"
        >
          <a
            class="text-gray-400 hover:text-violet-500"
            href="https://github.com/cremich/promptz"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="lucide lucide-github h-6 w-6"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
              />
              <path
                d="M9 18c-4.51 2-5-2-7-2"
              />
            </svg>
            <span
              class="sr-only"
            >
              GitHub
            </span>
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
